src/rotary-dialer-element.ts

```diff
@@ -91,9 +104,12 @@ export class RotaryDialerElement extends LitElement {
           <circle stroke="#fff" stroke-width="2" fill="#D8D8D8" cx="133" cy="133" r="72" />
           <path
             class=${classMap(this.classes)}
+            @transitionstart="${() => {
+              this.dialStartCount = this.fireOnce(this.dialStartCount, 'dial');
+            }}"
             @transitionend="${() => {
               this.removeDialerAnim();
-              this.dispatchEvent(new CustomEvent('dial', { detail: { digit: this.digit } }));
+              this.dialEndCount = this.fireOnce(this.dialEndCount, 'dial-end');
```

Comment:
How about looking at the CSS class to determine whether it's the middle or the end of the dial event?